docs/searching.rst

```diff
@@ -0,0 +1,348 @@
+a query which wil take a long time to execute and you want your script to do other things while waiting for the query
+to return.  Here's how we execute the device query from the last example asynchronously::
+
+    # create and refine a device query
+    >>> from cbc_sdk.platform import Device
+    >>> device_query = api.select(Device).where('avStatus:AV_ACTIVE').set_os(["WINDOWS"])
+
+    # now execute it
+    future = device_query.execute_async()
+
+    # await the results
+    device_list = future.result()
+
+The ``execute_async()`` method returns a standard ``concurrent.futures.Future`` object, and that ``Future``'s
+``result()`` method will return a list with the results of the query.
+
+Faceting
+--------
+
+Facet search queries return statistical information indicating the relative weighting of the requested values as per
+the specified criteria.  Only certain query types support faceting.
+
+Simple Faceting
+***************
+
+Simple faceting is built into certain queries, allowing you to generate a summary on certain fields of all objects that
+match the query. To perform this, create and refine a query object as you would normally, then call the ``facets()``
+method on the query, passing it the names of the fields you want to facet on.
+
+Here is an example for USB devices::
+
+    >>> from cbc_sdk.endpoint_standard import USBDevice
+    >>> usb_devices = api.select(USBDevice).set_statuses(['APPROVED'])
+    >>> facet_data = usb_devices.facets(['vendor_name', 'product_name'])
+
+This facet query might produce data that looks like this:
+
+.. code-block:: json
+
+    [
+        {
+            "field": "vendor_name",
+            "values": [
+                {
+                    "id": "Generic",
+                    "name": "Generic",
+                    "total": 2
+                },
+                {
+                    "id": "Kingston",
+                    "name": "Kingston",
+                    "total": 2
+                }
+            ]
+        },
+        {
+            "field": "product_name",
+            "values": [
+                {
+                    "id": "DataTraveler 3.0",
+                    "name": "DataTraveler 3.0",
+                    "total": 2
+                },
+                {
+                    "id": "Mass Storage",
+                    "name": "Mass Storage",
+                    "total": 2
+                }
+            ]
+        }
+    ]
+
+Facet Queries
+*************
+
+More complex facet queries are performed by creating a query *on* a facet type, then refining it as usual, then getting
+the results from the query::
+
+    >>> from cbc_sdk.endpoint_standard import EnrichedEventFacet
+    >>> query = api.select(EnrichedEventFacet).where(process_pid=1000)
+
+Facet queries have two types of special criteria that may be set. One is the ``range`` type which is used to specify
```

Comment:
Range is not faceting specific and should be called out as it's own section this is particularly important for Alerts and Processes

Reply:
How do you suggest this should be done?

Reply:
We should make a section on ranges separate from Facets that explains the set_time_range functions we have in a few queries now

Reply:
I may have something for this.